### ISSUE TITLE
Pwsh still corrupts environment variable on Linux Arm 64

### ISSUE THREAD

**Issue by SleepyFoxStudio** [PowerShell/PowerShell#24693]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

use the following docker file building on a Linux Arm machine (I'm using AWS Graviton EC2)

<pre>FROM mcr.microsoft.com/dotnet/sdk:9.0.101-noble-${TARGETARCH}${TARGETVARIANT}
ARG TARGETPLATFORM
ARG TARGETARCH
ARG TARGETVARIANT
ARG BUILDVARIANT
ARG ARM64ORX64
RUN printf "I'm building for TARGETPLATFORM=${TARGETPLATFORM}" \
&& printf ", TARGETVARIANT=${TARGETVARIANT}" \
&& printf ", TARGETARCH=${TARGETARCH}" \
&& printf ", BUILDVARIANT=${BUILDVARIANT} \n" \
&& printf ", ARM64ORX64=${ARM64ORX64} \n" \
    && printf "With uname -s : " && uname -s \
    && printf "and  uname -m : " && uname -m

RUN pwsh
RUN pwsh --version
RUN which pwsh
RUN echo $PATH
RUN ls -l /usr/bin/pwsh
RUN pwsh -Command 'Write-Host ${env:PSHOME}'
RUN pwsh -Command 'Write-Host ${env:PATH}'
RUN pwsh -Command 'pwsh -EncodedCommand VwByAGkAdABlAC0ASABvAHMAdAAgACcAaABlAGwAbABvACcA'
</pre>

and the following **Exec format error** is output

<pre> > [10/10] RUN pwsh -Command 'pwsh -EncodedCommand VwByAGkAdABlAC0ASABvAHMAdAAgACcAaABlAGwAbABvACcA':

[2024-12-18T13:30:25.569Z] 1.392 ResourceUnavailable: Program 'pwsh' failed to run: An error occurred trying to start process '/usr/share/powershell/.store/powershell.linux.arm64/7.5.0-preview.5/powershell.linux.arm64/7.5.0-preview.5/tools/net9.0/any/pwsh' with working directory '/'. Exec format errorAt line:1 char:1</pre>

[dockerlog.txt](https://github.com/user-attachments/files/18195619/dockerlog.txt)

this was originally raised via https://github.com/PowerShell/PowerShell/issues/24090 and I thought this was supposed to be resolved now

### Expected behavior

```console
[2024-12-18T13:27:43.017Z] #13 [10/10] RUN pwsh -Command 'pwsh -EncodedCommand VwByAGkAdABlAC0ASABvAHMAdAAgACcAaABlAGwAbABvACcA'


should output "hello"
```

### Actual behavior

```console
[2024-12-18T13:30:25.569Z] #13 1.392 ResourceUnavailable: Program 'pwsh' failed to run: An error occurred trying to start process '/usr/share/powershell/.store/powershell.linux.arm64/7.5.0-preview.5/powershell.linux.arm64/7.5.0-preview.5/tools/net9.0/any/pwsh' with working directory '/'. Exec format errorAt line:1 char:1
```

### Environment data

```powershell
RUN pwsh $PSVersionTable

[2024-12-19T10:04:36.769Z] #11 0.636 PowerShell 7.5.0-preview.5
```

**Comment by mklement0** [comment]
The problem seems to be that in your case the path contains the following segment:

`/.store/powershell.linux.arm64/`

whereas the fix in https://github.com/PowerShell/PowerShell/pull/24228/files is in effect looking for:

`/.store/powershell/`

**Comment by SleepyFoxStudio** [comment]
You're right, but I have not done anything special just using the latest image. (_mcr.microsoft.com/dotnet/sdk:9.0.101-noble-${TARGETARCH}${TARGETVARIANT}_)

I dont believe it's anything i have done to make it change the path to 

/.store/powershell.linux.arm64/

but i'm happy to test any changes you might think will fix it.
